bridge-history-api/internal/controller/history_controller.go

```diff
@@ -63,10 +88,53 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
 		types.RenderFailure(ctx, types.ErrParameterInvalidNo, err)
 		return
 	}
-	result, err := c.historyLogic.GetTxsByHashes(ctx, req.Txs)
-	if err != nil {
-		types.RenderFailure(ctx, types.ErrGetTxsByHashFailure, err)
+
+	if len(req.Txs) > 10 {
+		types.RenderFailure(ctx, types.ErrParameterInvalidNo, errors.New("the number of hashes in the request exceeds the allowed maximum"))
 		return
 	}
-	types.RenderSuccess(ctx, &types.ResultData{Result: result, Total: 0})
+
+	// deduplicate
+	hashesMap := make(map[string]struct{})
```

Comment:
```suggestion
	hashesMap := make(map[string]struct{}, len(req.Txs))
```

Reply:
changed in [8b93d0e](https://github.com/scroll-tech/scroll/pull/966/commits/8b93d0ea762262922636710ded81bd8fa653f5da)